java/src/main/java/com/genexus/db/driver/GXPreparedStatement.java

```diff
@@ -1376,9 +1376,23 @@ public void setBLOBFile(int index, String fileName, boolean isMultiMedia) throws
 			if (con.getContext() != null)
 			{
 				com.genexus.internet.HttpContext webContext = (HttpContext) con.getContext().getHttpContext();
-				if((webContext != null) && (webContext instanceof com.genexus.webpanels.HttpContextWeb))
+				if(webContext != null)
 				{
-					fileName = ((com.genexus.webpanels.HttpContextWeb)webContext).getRealPath(fileName);
+					if (webContext instanceof com.genexus.webpanels.HttpContextWeb) {
+						fileName = ((com.genexus.webpanels.HttpContextWeb) webContext).getRealPath(fileName);
+					}
+					else
+					{
+						if (!webContext.getDefaultPath().isEmpty() && !fileName.startsWith("http") && ! new File(fileName).isAbsolute())
```

Comment:
Some comments: 
1. Shouldn't we check  if it is an external (http) resource in first place? 

2. fileName.startsWith("http") seems to be fragile. 
ex: httpmyImage.png. 
We cloud check by full schema name: http:// or https://